cloud/gcp/deploy/website.go

```diff
@@ -18,13 +18,346 @@ package deploy
+		Name:        pulumi.String(subDomain),
+		ManagedZone: pulumi.String(managedZone.Name),
+		Type:        pulumi.String("A"),
+		Rrdatas:     pulumi.StringArray{ip.Address},
+		Ttl:         pulumi.IntPtr(300),
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = dns.NewRecordSet(ctx, "www-cdn-dns-record", &dns.RecordSetArgs{
+		Name:        pulumi.String(fmt.Sprintf("www.%s", subDomain)),
+		ManagedZone: pulumi.String(managedZone.Name),
+		Type:        pulumi.String("A"),
+		Rrdatas:     pulumi.StringArray{ip.Address},
+		Ttl:         pulumi.IntPtr(300),
+	})
+	if err != nil {
+		return err
+	}
+
+	// The certificate will use Load Balancer authorization (as opposed to DNS auth).
+	sslCert, err := certificatemanager.NewCertificate(ctx, "cdn-cert", &certificatemanager.CertificateArgs{
+		Scope: pulumi.String("DEFAULT"),
+		Managed: certificatemanager.CertificateManagedArgs{
+			Domains: pulumi.StringArray{
+				// Removing trailing dot (root zone), it's unsupported by certificate manager
+				pulumi.String(strings.TrimSuffix(subDomain, ".")),
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	certMap, err := certificatemanager.NewCertificateMapResource(ctx, "cert-map", &certificatemanager.CertificateMapResourceArgs{})
+	if err != nil {
+		return err
+	}
+
+	_, err = certificatemanager.NewCertificateMapEntry(ctx, "default", &certificatemanager.CertificateMapEntryArgs{
+		Name:        pulumi.String("cdn-cert-map-entry"),
+		Description: pulumi.String("CDN Certificate Map Entry"),
+		Map:         certMap.Name,
+		Certificates: pulumi.StringArray{
+			sslCert.ID(),
+		},
+		Matcher: pulumi.String("PRIMARY"),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create an HTTP proxy to route requests to the URLMap.
+	// https://www.pulumi.com/registry/packages/gcp/api-docs/compute/targethttpsproxy/#target-https-proxy-certificate-manager-certificate
+	httpsProxy, err := compute.NewTargetHttpsProxy(ctx, "http-proxy", &compute.TargetHttpsProxyArgs{
+		// CertificateManagerCertificates: pulumi.StringArray{pulumi.Sprintf("//certificatemanager.googleapis.com/%v", sslCert.ID())},
+		CertificateMap: pulumi.Sprintf("//certificatemanager.googleapis.com/%v", certMap.ID()),
+		UrlMap:         httpsUrlMap.SelfLink,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create a GlobalForwardingRule rule to route requests to the HTTP proxy.
+	_, err = compute.NewGlobalForwardingRule(ctx, "http-forwarding-rule", &compute.GlobalForwardingRuleArgs{
+		IpAddress:  ip.Address,
+		IpProtocol: pulumi.String("TCP"),
+		PortRange:  pulumi.String("443"),
+		Target:     httpsProxy.SelfLink,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Invalid the CDN Cache
+	local.NewCommand(ctx, "invalidate-cache", &local.CommandArgs{
+		// Only run on update, the cache is already empty on first creation
+		Create: pulumi.String(""),
+		Update: pulumi.Sprintf("gcloud compute url-maps invalidate-cdn-cache %s --path \"/*\"", httpsUrlMap.Name),
+	})
```

Comment:
Do we need to handle an error here or does it bubble up from pulumi's deployment engine anyway?